src/gordon_gcp/plugins/enricher.py

```diff
@@ -156,28 +156,50 @@ def __init__(self, config, http_client, success_channel, error_channel):
                f'{exception.__class__.__name__}: {exception}')
         raise exceptions.GCPGordonError(msg)
 
-    async def _create_A_rrecord(self, event_data, instance_data, msg_logger):
-        dns_zone = self.config['dns_zone']
-        hostname = event_data['resourceName'].split('/')[-1]
-        access_configs = instance_data['networkInterfaces'][0]['accessConfigs']
-        external_ip = access_configs[0]['natIP']
-
-        name = '.'.join([hostname, dns_zone])
-        msg_logger.debug(f'Creating A record: {name} -> {external_ip}')
+    def _create_A_rrecord(self, fqdn, external_ip, default_ttl, msg_logger):
+        msg_logger.debug(f'Creating A record: {fqdn} -> {external_ip}')
         return {
-            'name': name,
+            'name': fqdn,
             'rrdatas': [external_ip],
             'type': 'A',
-            'ttl': self.config['default_ttl']
+            'ttl': default_ttl
         }
 
-    async def _create_rrecords(self, event_data, msg_logger):
+    async def _get_full_data(self, event_data, msg_logger):
         instance_data = await self._poll_for_instance_data(
             event_data['resourceName'], msg_logger)
 
-        a_record = await self._create_A_rrecord(event_data, instance_data,
-                                                msg_logger)
-        return [a_record]
+        dns_zone = self.config['dns_zone']
+        hostname = event_data['resourceName'].split('/')[-1]
+        fqdn = '.'.join([hostname, dns_zone])
+
+        interfaces = instance_data['networkInterfaces']
+        external_ip = interfaces[0]['accessConfigs'][0]['natIP']
+        internal_ip = None
+        if len(interfaces) > 1:
+            internal_ip = interfaces[1]['accessConfigs'][0]['natIP']
+
+        default_ttl = self.config['default_ttl']
```

Comment:
I think this didn't change with this PR, but this can cause a runtime KeyError, since it is required but not validated.